.github/workflows/aks.yaml

```diff
@@ -128,13 +115,54 @@ jobs:
             --resource-group ${{ env.name }} \
             --name ${{ env.name }}
 
-      - name: Install Cilium
+      - name: Create kubeconfig and load it in configmap
+        run: |
+          .github/get-kubeconfig.sh
+          kubectl create configmap cilium-cli-kubeconfig -n kube-system --from-file kubeconfig
+
+      - name: Load cilium install script in configmap
         run: |
-          cilium install \
-            --cluster-name=${{ env.name }} \
-            --azure-resource-group ${{ env.name }} \
-            --wait=false \
-            --config monitor-aggregation=none
+          kubectl create configmap cilium-cli-test-script-install -n kube-system --from-file=in-cluster-test-script.sh=.github/in-cluster-test-scripts/aks-install.sh
+
+      - name: Set up Azure-specific CLI variables since `az` is not available in-cluster
+        id: az
+        run: |
+          # Derive subscription ID from credentials
+          AZURE_SUBSCRIPTION_ID=$(az account show --query "id" --output tsv)
+
+          # Derive AKS node resource group from created cluster
+          AZURE_NODE_RESOURCE_GROUP=$(az aks show --resource-group ${{ env.name }} --name ${{ env.name }} --query "nodeResourceGroup" --output tsv)
+
+          # Create Service Principal with minimal privileges over the AKS node resource group
+          AZURE_SERVICE_PRINCIPAL=$(az ad sp create-for-rbac --scopes /subscriptions/${AZURE_SUBSCRIPTION_ID}/resourceGroups/${AZURE_NODE_RESOURCE_GROUP} --output json --only-show-errors)
+          TENANT_ID=$(echo ${AZURE_SERVICE_PRINCIPAL} | jq -r '.tenant')
+          CLIENT_ID=$(echo ${AZURE_SERVICE_PRINCIPAL} | jq -r '.appId')
+          CLIENT_SECRET=$(echo ${AZURE_SERVICE_PRINCIPAL} | jq -r '.password')
+
+          echo ::set-output name=subscription-id::${AZURE_SUBSCRIPTION_ID}
+          echo ::set-output name=node-resource-group::${AZURE_NODE_RESOURCE_GROUP}
+          echo ::set-output name=tenant-id::${TENANT_ID}
+          echo ::set-output name=client-id::${CLIENT_ID}
+          echo ::add-mask::${CLIENT_SECRET}
+          echo ::set-output name=client-secret::${CLIENT_SECRET}
+
+      - name: Create cilium-cli install job
+        run: |
+          helm install .github/cilium-cli-test-job-chart \
+            --generate-name \
+            --set job_name=cilium-cli-install \
+            --set test_script_cm=cilium-cli-test-script-install \
+            --set tag=${{ steps.vars.outputs.sha }} \
+            --set cluster_name=${{ env.clusterName }} \
```

Comment:
This env variable is incorrect for AKS and most likely useless in this context. I will check tomorrow.

Reply:
Yep, it's now useless: cluster name is unused since we bypass `az` commands. It was used for deriving the AKS node resource group from the resource group name and cluster name via `aks show`.